src/main/java/com/google/devtools/build/lib/runtime/UiEventHandler.java

```diff
@@ -909,16 +889,19 @@ private synchronized boolean timeBasedRefresh() {
     if (!stateTracker.progressBarTimeDependent()) {
       return false;
     }
+    // Don't do more updates than are requested through events when there is no cursor control.
+    if (!cursorControl){
```

Comment:
Is this check necessary? I didn't follow the code entirely but this method seems like it should be entirely independent of whether or not there is cursor control.

(I didn't follow the code that closely so feel free to correct me if I'm wrong).

Reply:
The time-based refresh is on top of all events the `UiEventHandler` gets. I thought it was strange that the previous logic of updating every 20 seconds (without cursor control) interfered with the logic of the [ActionStatusReporter](https://github.com/bazelbuild/bazel/blob/master/src/main/java/com/google/devtools/build/lib/actions/ActionExecutionStatusReporter.java#L261) which is also mentioned in the [docs](https://bazel.build/reference/command-line-reference#flag--progress_report_interval), while in reality, the updates were coming in more often.

On the other hand, when there is cursor control, updating every second seems desirable, as it expected to see the second increments.